src/meson.build

```diff
@@ -56,6 +68,38 @@ libnvme_dep = declare_dependency(
     link_with: libnvme,
 )
 
+libnvme_mi_shared = shared_library(
+    'nvme-mi', # produces libnvme-mi.so
+    mi_sources,
+    version: libnvme_version,
+    soversion: '1',
+    link_args: ['-Wl,--version-script=' + mi_version_script_arg],
+    dependencies: mi_deps,
+    link_depends: mi_mapfile,
+    include_directories: incdir,
+    install: true,
+)
+
+libnvme_mi_static = static_library(
```

Comment:
Please adopt the same strategy how to build libraries as we have in mainline. Klaus fixed up my mess I did with the meson.build file. See commit:  c874f6697578 ("meson: use library() and rely on user settings")

Reply:
Ah, I'd based this on the earlier meson defs, which had both. I'll make this consistent with the current `meson.build`